code/model/DMSDocument.php

```diff
@@ -1341,14 +1348,32 @@ public function index(SS_HTTPRequest $request)
                         return $path;
                     }
 
+                    // set fallback if no config nor file-specific value
+                    $disposition = 'attachment';
+
+                    // file-specific setting
+                    if ($doc->DownloadBehavior == 'open') {
+                        $disposition = 'inline';
+                    } else if ($doc->DownloadBehavior == 'download') {
+                        $disposition = 'attachment';
+
+                    // no file preference configured, hand over to defaults
+                    } else {
```

Comment:
@dhensby Can you please explain why? Without this, the default functionality will not work.